dot/state/block.go

```diff
@@ -250,6 +250,37 @@ func (bs *BlockState) GetHashByNumber(num uint) (common.Hash, error) {
 	return common.NewHash(bh), nil
 }
 
+// GetBlockHashesBySlot gets all blocks that were produced in given slot.
+func (bs *BlockState) GetBlockHashesBySlot(slotNum uint64) ([]common.Hash, error) {
+	highestFinalisedHash, err := bs.GetHighestFinalisedHash()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get highest finalised hash: %w", err)
+	}
+
+	descendants, err := bs.bt.GetAllDescendants(highestFinalisedHash)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get descendants: %w", err)
+	}
+
+	blocksWithGivenSlot := []common.Hash{}
+
+	for _, desc := range descendants {
+		descSlot, err := bs.GetSlotForBlock(desc)
+		if errors.Is(err, types.ErrGenesisHeader) {
+			continue
+		}
+		if err != nil {
+			return nil, fmt.Errorf("could not get slot for block %s: %w", desc, err)
```

Comment:
`desc` is to be wrapped by `GetSlotForBlock`, not here

```suggestion
			return nil, fmt.Errorf("could not get slot for block: %w", err)
```